storage/local/storage.go

```diff
@@ -391,12 +358,73 @@ func (s *memorySeriesStorage) FingerprintsForLabelMatchers(labelMatchers metric.
 		}
 		result = intersection
 	}
+	return result
+}
+
+// MetricsForLabelMatchers returns the metrics from storage that satisfy the given label matchers.
```

Comment:
The check for at least one matcher that doesn't match "" is in the parser. So we need to document here the behavior if all matchers match "".